src/mongocrypt-crypto-private.h

```diff
@@ -44,9 +44,19 @@ typedef struct {
 uint32_t
 _mongocrypt_calculate_ciphertext_len (uint32_t plaintext_len);
 
+/* _mongocrypt_fle2aead_calculate_ciphertext_len returns the required length of
+ * the ciphertext for _mongocrypt_fle2aead_do_encryption. */
+uint32_t
+_mongocrypt_fle2aead_calculate_ciphertext_len (uint32_t plaintext_len);
```

Comment:
I prefer to keep this as-is, but I am open to further discussion.

I consider `_mongocrypt_fle2aead_do_encryption` and `_mongocrypt_fle2aead_do_decryption` to be specific to FLE 2 since they follow the construction from [AEAD with CTR](https://docs.google.com/document/d/1eCU7R8Kjr-mdyz6eKvhNIDVmhyYQcAaLtTfHeK7a_vE/).

Using `fle2aead` is consistent with these functions in the MongoDB server: https://github.com/mongodb/mongo/blob/cf22bb8204da47c2d3696aabb2574077a66dcd66/src/mongo/crypto/aead_encryption.h#L99

AFAIK there is no opportunity for reuse currently.

IMO having `fle2` as part of the symbol makes it easier to track what applies to FLE 1 and FLE 2.